geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -381,19 +408,26 @@ public synchronized void close(SocketChannel socketChannel) {
     try (final ByteBufferSharing outputSharing = outputBufferVendor.open(1, TimeUnit.MINUTES)) {
       final ByteBuffer myNetData = outputSharing.getBuffer();
 
-      if (!engine.isOutboundDone()) {
-        ByteBuffer empty = ByteBuffer.wrap(new byte[0]);
-        engine.closeOutbound();
+      engine.closeOutbound();
+
+      SSLEngineResult result = null;
+
+      while (!engine.isOutboundDone()) {
+        final ByteBuffer empty = ByteBuffer.wrap(new byte[0]);
 
         // clear the buffer to receive a CLOSE message from the SSLEngine
         myNetData.clear();
 
         // Get close message
-        SSLEngineResult result = engine.wrap(empty, myNetData);
-
-        if (result.getStatus() != SSLEngineResult.Status.CLOSED) {
-          throw new SSLHandshakeException(
-              "Error closing SSL session.  Status=" + result.getStatus());
+        result = engine.wrap(empty, myNetData);
```

Comment:
Is it possible to get a `BUFFER_OVERFLOW` here?

Reply:
I don't think so since we are wrapping an empty buffer.

Reply:
But what is expected here is an SSL message, in theory they should be small enough to fit in whatever buffer we have.